ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/SecretsUrl.scala

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.tls
+
+import java.io.InputStream
+import java.net.URL
+import java.nio.file.{Files, Path}
+
+// This trait is not sealed so we can replace it with a fake in tests.
+trait SecretsUrl {
+  def openStream(): InputStream
+}
+
+object SecretsUrl {
+  def FromString(string: String): SecretsUrl = FromUrl(new URL(string))
```

Comment:
If this is a method, it should be named like:
```suggestion
  def fromString(string: String): SecretsUrl = FromUrl(new URL(string))
```

Reply:
I'm not convinced; I know it's a method but I want it to look like any other constructor.

If I change this one I'll also make the subclasses private and add wrappers along the lines of `def fromUrl(url: Url) = FromUrl(url)` so they're lowercase too.

Reply:
I agree with @miklos-da - to me `SecretsUrl.FromString(url)` looks like creation of a class named `FromString`, especially that we have other classes under `SecretsUrl`.
I would either changed the name to `fromString()` or add a constructor in the companion object:
```
  object FromUrl {
    def apply(string: String): FromUrl = FromUrl(new URL(string))
  }
  
  ```

Reply:
Made them all methods.